metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/AuthEnabledConnectorTableService.java

```diff
@@ -0,0 +1,125 @@
+                                                          final List<String> uris,
+                                                          final boolean prefixSearch) {
+        return delegate.getTableNames(context, uris, prefixSearch);
+    }
+
+    @Override
+    public List<QualifiedName> getTableNames(final ConnectorRequestContext context,
+                                             final QualifiedName name,
+                                             final String filter,
+                                             final Integer limit) {
+        authorize(MetacatContextManager.getContext(), name);
+        return delegate.getTableNames(context, name, filter, limit);
+    }
+
+    @Override
+    public void create(final ConnectorRequestContext context, final TableInfo resource) {
+        authorize(MetacatContextManager.getContext(), resource.getName());
+        delegate.create(context, resource);
+    }
+
+    @Override
+    public void update(final ConnectorRequestContext context, final TableInfo resource) {
+        authorize(MetacatContextManager.getContext(), resource.getName());
+        delegate.update(context, resource);
+    }
+
+    @Override
+    public void delete(final ConnectorRequestContext context, final QualifiedName name) {
+        authorize(MetacatContextManager.getContext(), name);
+        delegate.delete(context, name);
+    }
+
+    @Override
+    @SuppressFBWarnings
+    public boolean exists(final ConnectorRequestContext context, final QualifiedName name) {
+        authorize(MetacatContextManager.getContext(), name);
+        return delegate.exists(context, name);
+    }
+
+    @Override
+    public List<TableInfo> list(final ConnectorRequestContext context, final QualifiedName name,
+                                @Nullable final QualifiedName prefix,
+                                @Nullable final Sort sort, @Nullable final Pageable pageable) {
+        authorize(MetacatContextManager.getContext(), name);
+        return delegate.list(context, name, prefix, sort, pageable);
+    }
+
+    @Override
+    public List<QualifiedName> listNames(final ConnectorRequestContext context, final QualifiedName name,
+                                         @Nullable final QualifiedName prefix,
+                                         @Nullable final Sort sort, @Nullable final Pageable pageable) {
+        authorize(MetacatContextManager.getContext(), name);
+        return delegate.listNames(context, name, prefix, sort, pageable);
+    }
+
+    @Override
+    public void rename(final ConnectorRequestContext context,
+                       final QualifiedName oldName,
+                       final QualifiedName newName) {
+        authorize(MetacatContextManager.getContext(), oldName);
+        delegate.rename(context, oldName, newName);
+    }
+
+    private void authorize(final MetacatRequestContext context, final QualifiedName resource) {
```

Comment:
Adding to the previous comment, using such a framework and extending it would make it so you wouldn't need to have basically the same method and logic duplicated three times across classes which is not a great pattern for code structure or maintainability.